src/Chapter3.hs

```diff
@@ -910,6 +1000,17 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+    append (Gold a) (Gold b) = Gold $ a + b
+
+instance Append ([a]) where
+    append xs ys = xs ++ ys
+
+instance (Append a) => Append (Maybe a) where
+    append (Just x) (Just y) = Just $ append x y
+    append _ _ = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality it would be more beneficial is this function returned `Just` when at least one argument is `Just`.